website/api/zkvm/zkvm-specification.md

```diff
@@ -35,22 +35,49 @@ preemption, so there is also no support for atomic instructions. If the
 execution raises an exception such as an unaligned access, the execution
 terminates without executing exception handlers.
 
+## The User/Kernel Split
+
+Starting with the v2 circuit, the RISC Zero zkVM introduces distinct user and
+kernel modes. Guest code running on the zkVM is divided into two components: the
+kernel and user-space programs, both executing the rv32im instruction set. The
+kernel functions as an operating system kernel, while user programs run in a
+separate user execution mode. This separation ensures that the kernel's resources
+remain protected from user-space code.
+
+When executing guest code, the zkVM first runs the kernel, which is responsible
+for initializing system resources. The specific tasks performed by the kernel
+depend on its implementation, but its primary role is to prepare the environment
+for user programs. Once initialization is complete, control is handed over to
+the user program for execution.
```

Comment:
Today, the kernel's purpose is to just provide emulation support for the v1 ABI.

Maybe we should drop this section.

Reply:
yeah, it's definitely independent and things like this are general knowledge. so maybe i can just link to an os dev wiki page